### ISSUE TITLE
0.18.0: tests no longer work offline

### ISSUE THREAD

**Issue by mgorny** [15r10nk/inline-snapshot#165]
Starting with 0.18.0, the tests no longer work offline. I'm seeing the following test failures:

```pytb
ERROR tests/adapter/test_dataclass.py::test_attrs_default_value - Failed: some snapshots in this test have incorrect values.
ERROR tests/adapter/test_dataclass.py::test_attrs_field_repr - Failed: some snapshots in this test have incorrect values.
ERROR tests/test_pydantic.py::test_pydantic_create_snapshot[pydantic>=2.0.0] - Failed: some snapshots in this test have incorrect values.
ERROR tests/test_pydantic.py::test_pydantic_create_snapshot[pydantic<2.0.0] - Failed: some snapshots in this test have incorrect values.
ERROR tests/test_pydantic.py::test_pydantic_field_repr[pydantic>=2.0.0] - Failed: some snapshots in this test have incorrect values.
ERROR tests/test_pydantic.py::test_pydantic_field_repr[pydantic<2.0.0] - Failed: some snapshots in this test have incorrect values.
ERROR tests/test_pydantic.py::test_pydantic_default_value[pydantic>=2.0.0] - Failed: some snapshots in this test have incorrect values.
ERROR tests/test_pydantic.py::test_pydantic_default_value[pydantic<2.0.0] - Failed: some snapshots in this test have incorrect values.
```

They are all failing with confusing error messages, because `uv` attempts fetching packages:
```
run> uv run --with=pydantic<2.0.0 python -m pytest --inline-snapshot=update
stdout:

stderr:
error: Failed to fetch: `https://pypi.org/simple/pydantic/`
  Caused by: Could not connect, are you offline?
  Caused by: Request failed after 3 retries
  Caused by: error sending request for url (https://pypi.org/simple/pydantic/)
  Caused by: client error (Connect)
  Caused by: dns error: failed to lookup address information: Temporary failure in name resolution
  Caused by: failed to lookup address information: Temporary failure in name resolution
```

Could you please make the tests don't call `uv`, and instead use installed test dependencies when they are available?

**Comment by 15r10nk** [comment]
Yes, I will make it optional. It is just really useful if I run my tests locally and have to test different versions of the same package.

**Comment by mgorny** [comment]
Thank you!

**Comment by 15r10nk** [comment]
solved in 0.18.1
